File: C/Grace.c
#include <stdio.h>

/*
 This program writes its own source code to a file.
*/

#define STR(X) #X
#define XSTR(X) STR(X)
#define FT(X) int main(){FILE *f=fopen("Grace_kid.c","w");fprintf(f,"#include <stdio.h>\n\n/*\n This program writes its own source code to a file.\n*/\n\n#define STR(X) #X\n#define XSTR(X) STR(X)\n#define FT(X) "X"\n\nFT(XSTR(FT(X)))\n");fclose(f);}

FT(XSTR(FT(X)))

File: Python/Grace.py
"""
 This program writes its own source code to a file.
"""

name = "Grace_kid.py"
modes = "w"
code = '"""\n This program writes its own source code to a file.\n"""\n\nname = "{0}"\nmodes = "w"\ncode = {1}\nf = open(name, modes)\nf.write(code.format(name, repr(code)))\nf.close()\n'
f = open(name, modes)
f.write(code.format(name, repr(code)))
f.close()
